From 658957c4c6dcc373dadb38348a17f46f406ae250 Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Mon, 5 Mar 2018 22:32:26 -0800
Subject: [PATCH 337/375] mmc: cavium_mmc: Configure sample command and data
 directly

no need to scale to SCLK for thunderX platform.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
---
 drivers/mmc/host/cavium.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/cavium.c b/drivers/mmc/host/cavium.c
index 1a9f4fb07e52..6a4c76918e47 100644
--- a/drivers/mmc/host/cavium.c
+++ b/drivers/mmc/host/cavium.c
@@ -952,7 +952,6 @@ static int cvm_mmc_of_parse(struct device *dev, struct cvm_mmc_slot *slot)
 	u32 id, cmd_skew = 0, dat_skew = 0, bus_width = 0;
 	struct device_node *node = dev->of_node;
 	struct mmc_host *mmc = slot->mmc;
-	u64 clock_period;
 	int ret;
 
 	ret = of_property_read_u32(node, "reg", &id);
@@ -1000,11 +999,10 @@ static int cvm_mmc_of_parse(struct device *dev, struct cvm_mmc_slot *slot)
 	mmc->f_min = 400000;
 
 	/* Sampling register settings, period in picoseconds */
-	clock_period = 1000000000000ull / slot->host->sys_freq;
 	of_property_read_u32(node, "cavium,cmd-clk-skew", &cmd_skew);
 	of_property_read_u32(node, "cavium,dat-clk-skew", &dat_skew);
-	slot->cmd_cnt = (cmd_skew + clock_period / 2) / clock_period;
-	slot->dat_cnt = (dat_skew + clock_period / 2) / clock_period;
+	slot->cmd_cnt = cmd_skew;
+	slot->dat_cnt = dat_skew;
 
 	return id;
 }
-- 
2.14.1

